src/main/java/com/exasol/adapter/dynamodb/DynamodbAdapter.java

```diff
@@ -31,51 +37,41 @@
 
 	@Override
 	public CreateVirtualSchemaResponse createVirtualSchema(final ExaMetadata exaMetadata,
-			final CreateVirtualSchemaRequest request) {
-		final ColumnMetadata.Builder colBuilder = new ColumnMetadata.Builder();
-		colBuilder.name("isbn");
-		colBuilder.type(DataType.createVarChar(100, DataType.ExaCharset.ASCII));
-		final List<ColumnMetadata> cols = List.of(colBuilder.build());
-		final List<TableMetadata> tables = List.of(new TableMetadata("testTable", "", cols, ""));
-		final SchemaMetadata remoteMeta = new SchemaMetadata("", tables);
-		return CreateVirtualSchemaResponse.builder().schemaMetadata(remoteMeta).build();
+			final CreateVirtualSchemaRequest request) throws AdapterException {
+		try {
+			final SchemaMappingDefinition schemaMappingDefinition = getSchemaMappingDefinition(request);
+			final SchemaMetadata schemaMetadata = SchemaMappingDefinitionToSchemaMetadataConverter
+					.convert(schemaMappingDefinition);
+			return CreateVirtualSchemaResponse.builder().schemaMetadata(schemaMetadata).build();
+		} catch (final IOException e) {
+			throw new AdapterException("Unable create Virtual Schema \"" + request.getVirtualSchemaName()
+					+ "\". Cause: \"" + e.getMessage(), e);// NOSONAR
+		}
+	}
+
+	private SchemaMappingDefinition getSchemaMappingDefinition(final AdapterRequest request)
+			throws AdapterException, IOException {
+		final AdapterProperties adapterProperties = new AdapterProperties(
+				request.getSchemaMetadataInfo().getProperties());
+		final DynamodbAdapterProperties dynamodbAdapterProperties = new DynamodbAdapterProperties(adapterProperties);
+		final File path = dynamodbAdapterProperties.getMappingDefinition();
+		if (!path.exists()) {
+			throw new AdapterException(String.format("the specified mapping file (%s) could not be found.", path));
```

Comment:
Why do you use `String.format` in this place where simple string concatenation would achieve the same result?

Reply:
well I though it would be cool...